qiskit/primitives/containers/bit_packing.py

```diff
@@ -0,0 +1,510 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2024.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""
+Helper functions for packing and unpacking bit arrays.
+"""
+
+from __future__ import annotations
+from typing import Literal, Any
+from collections.abc import Sequence
+from numbers import Integral
+from dataclasses import dataclass
+from functools import lru_cache
+
+from numpy.typing import ArrayLike, NDArray
+import numpy as np
+
+
+def pack_bits(
+    array: ArrayLike[np.bool_],
```

Comment:
```suggestion
    array: ArrayLike,
```

> qiskit/primitives/containers/bit_packing.py:29:11: E1136: Value 'ArrayLike' is unsubscriptable (unsubscriptable-object)

https://dev.azure.com/qiskit-ci/qiskit/_build/results?buildId=60168&view=logs&j=a15d9aca-d75c-55b1-7f76-889cfe884cf0&t=154f6391-7dd7-5733-2230-2f10f0798b76&l=619